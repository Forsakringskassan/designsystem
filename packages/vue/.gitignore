# vite dev
/src/vite-dev/app.vue
