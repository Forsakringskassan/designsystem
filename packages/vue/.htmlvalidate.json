{
    "extends": ["@fkui/vue:recommended"],
    "plugins": ["./htmlvalidate"],
    "transform": {
        "^.*\\.spec.ts$": "html-validate-vue:html"
    }
}
