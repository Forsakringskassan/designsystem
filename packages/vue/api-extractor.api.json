{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/api.d.ts",
    "dtsRollup": {
        "enabled": false
    },
    "messages": {
        "compilerMessageReporting": {
            "default": {
                "logLevel": "none"
            }
        }
    }
}
