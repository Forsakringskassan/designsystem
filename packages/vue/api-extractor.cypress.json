{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/cypress/index.d.ts",
    "compiler": {
        "tsconfigFilePath": "<projectFolder>/tsconfig.pageobjects.json"
    },
    "apiReport": {
        "reportFileName": "<unscopedPackageName>-cypress.api.md"
    },
    "docModel": {
        "enabled": true,
        "apiJsonFilePath": "<projectFolder>/temp/cypress.api.json"
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/cypress.d.ts"
    },
    "messages": {
        "extractorMessageReporting": {
            "ae-unresolved-link": {
                "logLevel": "none"
            }
        }
    }
}
