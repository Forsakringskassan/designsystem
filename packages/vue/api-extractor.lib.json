{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "compiler": {
        "tsconfigFilePath": "<projectFolder>/tsconfig.lib.json",
        "overrideTsconfig": {
            "moduleResolution": "node"
        }
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/index.d.ts"
    },
    "messages": {
        "compilerMessageReporting": {
            "default": {
                "logLevel": "none"
            }
        }
    }
}
