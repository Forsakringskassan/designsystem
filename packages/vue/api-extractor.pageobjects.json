{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/pageobjects.d.ts",
    "compiler": {
        "tsconfigFilePath": "<projectFolder>/tsconfig.pageobjects.json"
    },
    "apiReport": {
        "reportFileName": "<unscopedPackageName>-pageobjects.api.md"
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/pageobject.d.ts"
    }
}
