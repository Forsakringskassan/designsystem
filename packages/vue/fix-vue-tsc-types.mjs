import fs from "node:fs/promises";
import path from "node:path";
import { glob } from "glob";

let n = 0;
console.group("Patching vue-tsc dts files (microsoft/rushstack#5146)");

const filenames = await glob("**/*.vue.d.ts", { cwd: "temp/types" });
const promises = filenames.map(async (filename) => {
    const filePath = path.join("temp/types", filename);
    const content = await fs.readFile(filePath, "utf-8");
    const updated = content.replace(
        /declare var (__VLS_\d+)/,
        "declare const $1",
    );
    if (content !== updated) {
        await fs.writeFile(filePath, updated, "utf-8");
        n++;
        console.log(filename);
    }
});
await Promise.all(promises);

console.groupEnd();
console.log(`${n} file${n === 1 ? "" : "s"} patches\n`);
