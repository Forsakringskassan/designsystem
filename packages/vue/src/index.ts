/* include typescript declarations for vite static asset handling (e.g. `?raw`) */
/// <reference types="vite/client" />

export * from "./components";
export {
    type UseSlotUtils,
    type WithoutInstance,
    useSlotUtils,
} from "./composables";
export {
    config,
    type FKUIConfig,
    FKUIConfigButtonOrder,
    type FKUIContext,
    type MaybeWithFKUIContext,
    setRunningContext,
} from "./config";
export * from "./internal-components";
export * from "./plugins";
export {
    type ComponentUnmountEvent,
    type ComponentValidityEvent,
    type ErrorItem,
    type GroupValidityEvent,
    type ListArray,
    type ListItem,
    type UnknownItem,
    ErrorData,
    ErrorViewData,
    FormErrorList,
    MenuAction,
} from "./types";
export * from "./utils";
