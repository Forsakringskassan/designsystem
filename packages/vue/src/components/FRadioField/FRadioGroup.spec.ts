import "html-validate/jest";
import { defineComponent, h } from "vue";
import { VueWrapper, mount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import { FRadioGroup, FRadioGroupField } from ".";

const defaultSlots = { label: "Label", default: "Content" };

function createWrapper({
    props = {},
    slots = {},
    attrs = {},
} = {}): VueWrapper {
    return mount(FRadioGroup, {
        attrs: { ...attrs },
        props: { id: "someId", name: "someName", ...props },
        slots: { ...slots },
    });
}

describe("snapshots", () => {
    it("should match snapshot with default classes and filled slots", () => {
        const wrapper = createWrapper({
            slots: {
                tooltip: "Tooltip",
                description: "Description",
                "error-message": "Error message",
                ...defaultSlots,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with horizontal layout", () => {
        const wrapper = createWrapper({ props: { isHorizontal: true } });
        expect(wrapper).toMatchSnapshot();
    });

    it("should match snapshot with indentation", async () => {
        const wrapper = mount({
            components: { FRadioGroup, FRadioGroupField },
            template: /* HTML */ `
                <f-radio-group name="foo">
                    <template #default="{ indentClass }">
                        <div :class="indentClass">
                            <f-radio-group-field value="bar" />
                        </div>
                    </template>
                </f-radio-group>
            `,
        });

        await flushPromises();
        expect(wrapper.element).toMatchSnapshot();
    });

    it("should match snapshot with child input element having same name as provided", () => {
        const wrapper = mount(
            defineComponent({
                components: {
                    FRadioGroup,
                    FRadioGroupField,
                },
                render() {
                    return h(FRadioGroup, { name: "sameName" }, () => [
                        h(FRadioGroupField, {
                            value: "some value",
                        }),
                    ]);
                },
            }),
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe("attributes", () => {
    it("should pass attributes", () => {
        const wrapper = createWrapper({
            attrs: {
                disabled: true,
                required: true,
            },
        });
        const input = wrapper.get("fieldset");

        expect(input.attributes("disabled")).toBeDefined();
        expect(input.attributes("required")).toBeDefined();
    });
});

describe("events", () => {
    it("should pass listeners", async () => {
        const foobar = jest.fn();
        const wrapper = createWrapper({
            attrs: { onFoobar: foobar },
        });
        const element = wrapper.get("fieldset");
        await element.trigger("foobar");
        expect(foobar).toHaveBeenCalled();
    });
});
