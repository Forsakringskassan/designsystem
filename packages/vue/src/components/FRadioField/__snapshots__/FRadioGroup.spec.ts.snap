// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should match snapshot with child input element having same name as provided 1`] = `
<fieldset id="fkui-vue-element-0001" class="fieldset">
  <legend class="label fieldset__label">
    <!--v-if-->
    <!--v-if-->
  </legend>
  <!--v-if-->
  <!--v-if-->
  <div class="fieldset__content">
    <div class="radio-button"><input id="fkui-vue-element-0002" type="radio" class="radio-button__input" name="sameName" value="some value"><label class="radio-button__label" for="fkui-vue-element-0002">
        <!--v-if-->
      </label></div>
  </div>
</fieldset>
`;

exports[`snapshots should match snapshot with default classes and filled slots 1`] = `
<fieldset
  class="fieldset"
  id="someId"
>
  <legend
    class="label sr-only"
  >
    
    
    Label
    
    
    <!--v-if-->
    
    
    Description
    
    
    
    
    Error message
    
    
  </legend>
  <!--v-if-->
  
  <div
    class="label"
  >
    <span
      aria-hidden="true"
    >
      
      
      Label
      
      
    </span>
  </div>
  
  
  Tooltip
  
  
  <div
    aria-hidden="true"
    class="label fieldset__label"
  >
    
    
    Description
    
    
    
    
    Error message
    
    
  </div>
  
  <div
    class="fieldset__content"
  >
    
    
    Content
    
    
  </div>
</fieldset>
`;

exports[`snapshots should match snapshot with horizontal layout 1`] = `
<fieldset id="someId" class="fieldset">
  <legend class="label fieldset__label">
    <!--v-if-->
    <!--v-if-->
  </legend>
  <!--v-if-->
  <!--v-if-->
  <div class="fieldset__content"></div>
</fieldset>
`;

exports[`snapshots should match snapshot with indentation 1`] = `
<fieldset
  class="fieldset radio-button-group"
  id="fkui-vue-element-0001"
>
  <legend
    class="label fieldset__label"
  >
    
    
    <!--v-if-->
    
    
    
    <!--v-if-->
    
  </legend>
  <!--v-if-->
  <!--v-if-->
  <div
    class="fieldset__content radio-button-group__content"
  >
    
    
    <div
      class="indent"
    >
      <div
        class="radio-button"
      >
        <input
          class="radio-button__input"
          id="fkui-vue-element-0002"
          name="foo"
          type="radio"
          value="bar"
        />
        <label
          class="radio-button__label"
          for="fkui-vue-element-0002"
        >
          
          
          <!--v-if-->
        </label>
      </div>
    </div>
    
    
  </div>
</fieldset>
`;
